app/views/user_products/_form.html.erb

```diff
@@ -17,6 +17,52 @@
 		    </div>
 		  <% end %>
 
+      <% if @user_product.id %>
+        <div
+		  		id="<%= "image-modal-holder-#{@user_product.id}" %>"
+		  		class="modal hide fade"
+		  		role="dialog"
+		  		aria-labelledby="basicModal"
+		  		aria-hidden="true"
+		  	>
+		  		<div class="modal-dialog" role="document">
+		  			<div class="modal-content">
+		  				<div class="modal-header">
+		  					<h4> Imagen del Producto </h4>
+		  					<button
+                  type="button"
+                  class="close"
+                  data-dismiss="modal"
+                  aria-hidden="true"
+                >
+                  ×
+                </button>
+		  				</div>
+		  				<div class="modal-body">
+		  					<%= @user_product.product.image.attached? ?
```

Comment:
Podrías usar `delegate` para no romper la "Law of Demeter". sería algo como:

`delegate :image, to: :product, prefix: false, allow_nil: true`

y luego hacés: `@user_product.image.attached?`

https://devblast.com/b/rails-delegate-dont-break-the-law-of-demeter

Reply:
Listo!